doc/source/tutorials/filters.rst

```diff
@@ -0,0 +1,372 @@
+the same list of filters in reverse, producing the original data.
+
+When filtering the data tiles of an attribute, TileDB stores some
+necessary metadata in file ``__fragment_metadata.tdb``, such as the
+starting location of each filtered tile and the original tile size
+in the case of variable-length attributes (recall that the original tile
+size has fixed size for fixed-length attributes in *both* dense and
+sparse arrays).
+
+Filter options
+--------------
+
+Some filters have additional options that can be configured. For example, you
+can set the compression level as an option on the filters that perform
+compression.
+
+.. content-tabs::
+
+   .. tab-container:: cpp
+      :title: C++
+
+      To set the compression level to level 5 on a bzip2 compression filter:
+
+      .. code-block:: c++
+
+        Filter compression_bzip2(ctx, TILEDB_FILTER_BZIP2);
+        int level = 5;
+        compression_bzip2.set_option(TILEDB_COMPRESSION_LEVEL, &level);
+
+      You can also retrieve option values from filters:
+
+      .. code-block:: c++
+
+        int level_get;
+        compression_bzip2.get_option(TILEDB_COMPRESSION_LEVEL, &level_get);
+        // Now level_get == 5
+
+The options supported by each filter are documented below.
+
+Available filters
+-----------------
+
+TileDB supports a number of filters, and more will continue to be added in the
+future.
+
+Compression filters
+~~~~~~~~~~~~~~~~~~~
+
+There are several filters performing generic compression, which are the following:
+
+* ``TILEDB_FILTER_GZIP``: Compresses with `Gzip <http://www.zlib.net/>`__
+* ``TILEDB_FILTER_ZSTD``: Compresses with `Zstandard <http://facebook.github.io/zstd/>`__
+* ``TILEDB_FILTER_LZ4``: Compresses with `LZ4 <https://github.com/lz4/lz4>`__
+* ``TILEDB_FILTER_RLE``: Compresses with `run-length encoding <https://en.wikipedia.org/wiki/Run-length_encoding>`__
+* ``TILEDB_FILTER_BZIP2``: Compresses with `Bzip2 <http://www.bzip.org/>`__
+* ``TILEDB_FILTER_DOUBLE_DELTA``: Compresses with double-delta encoding
+
+All of these filters support one filter option to set the compression level,
+although some compressors such as RLE currently ignore the setting. The filter
+option is:
+
+* ``TILEDB_COMPRESSION_LEVEL`` (type ``int32_t``): The compression level to
+  use. Default: -1 (compressor-specific default).
+
+Byteshuffle
+~~~~~~~~~~~
+
+The filter ``TILEDB_FILTER_BYTESHUFFLE`` performs byte shuffling of data as a
+way to improve compression ratios. The
+`byte shuffle implementation <{tiledb_src_root_url}/external/include/blosc/shuffle.h>`_
+used by TileDB comes from the `Blosc <blosc.org>`_ project.
+
+The byte shuffling process rearranges the bytes of the input attribute cell
+values in a deterministic and reversible manner designed to result in long runs
+of similar bytes that can be compressed more effectively by a generic compressor
+than the original unshuffled elements. Typically this filter would not be used
```

Comment:
suggestion to use active voice: "is not used on it's own, but rather immediately before a compression filter"